newscoop/library/Newscoop/SmartyView.php

```diff
@@ -11,30 +11,28 @@
  */
 class SmartyView extends \Zend_View_Abstract
 {
-    /** @var Smarty */
-    protected $smarty;
-
-    /**
-     */
-    public function __construct()
-    {
-        $this->smarty = \CampTemplate::singleton();
-    }
-
     /**
      * Render template
      */
     public function _run()
     {
-        foreach ($this->getVars() as $key => $val) {
-            $this->smarty->assign($key, $val);
-        }
+        $container = \Zend_Registry::get('container');
+        $templatesService = $container->getService('newscoop.templates.service');
+        $request = $container->getService('request');
+        $language = $container->get('em')->getRepository('Newscoop\Entity\Language')->findOneByCode($request->getLocale());
 
-        $this->smarty->assign('view', $this);
-        $this->smarty->assign('gimme', $this->smarty->context());
```

Comment:
`gimme` is set already in ThemesService (in construct) and `view` is set manually few lines bellow.